pkg/recipes/terraform/install.go

```diff
@@ -78,5 +81,40 @@ func Install(ctx context.Context, installer *install.Installer, tfDir string) (s
 
 	logger.Info(fmt.Sprintf("Terraform latest version installed to: %q", execPath))
 
-	return execPath, nil
+	// Create a new instance of tfexec.Terraform with current Terraform installation path
+	tf, err := NewTerraform(ctx, tfDir, execPath)
```

Comment:
Why don't we do the retry in this function?

Reply:
The problem happens when we try to use the terraform executable (i.e. `tf`). So we don't want to retry the creation of it, just when it's first used